revokefs/main.c

```diff
@@ -529,6 +550,14 @@ main (int argc, char *argv[])
           exit (EXIT_FAILURE);
         }
 
+      if (conf.exit_with_fd)
+        {
+          g_autoptr (GThread) monitor_exit_fd = NULL;
+          g_autoptr (GError) error = NULL;
+
+          /* Setup thread monitoring on exit of the parent process that spawned this backend */
+          monitor_exit_fd = g_thread_new (NULL, monitor_exit_fd_thread, GINT_TO_POINTER (conf.exit_with_fd));
+        }
```

Comment:
Using a thread for this seems quite wasteful. We already have a read loop in the writer. Just pass both fds there and have it read from both (using poll()/select())

Reply:
The issue I faced with that was, reading on both fds in a single loop **blocks** on data being available on the writer fd. There was a [similar approach](https://github.com/flatpak/flatpak/commit/cd8900f0c9e5326216df963f482c62a28a46fac0) of what you are saying here.
We cannot find out whether the system-helper was killed until some data is available on writer's fd (and next iteration of the loop happens).
Probably this will only work when we set O_NONBLOCK on writer's fd.

Reply:
Yes, you can't read both fds at the same time, that is why I mentioned poll/select. That is how you support reading from several fds.

Reply:
Addressed in the recent round of iteration.